include/af/opencv3.h

```diff
@@ -0,0 +1,191 @@
+                }
+            } else {
+                output = af::array(w, h, dims);
+                for(unsigned d=0; d<dims; ++d) {
+                    output(af::span, af::span, d) = af::array(w, h, channels[d].ptr<float>(0));
+                }
+            }
+        } else {
+            throw af::exception("mat to array error ");
+        }
+    }
+}
+
+
+/*
+ * opencv -> arrayfire
+ */
+
+// conversion for cv::Mat
+static af::array array(const cv::Mat& input, bool transpose)
+{
+    af::array output;
+    if (input.empty()) { return output; }
+    cv::Mat tmp;
+    if (input.channels() == 1)
+        { input.convertTo(tmp, CV_32F); }
+    else if (input.channels() == 2)
+        { input.convertTo(tmp, CV_32FC2); }
+    else if (input.channels() == 3)
+        { input.convertTo(tmp, CV_32FC3); }
+    else if (input.channels() == 4)
+        { input.convertTo(tmp, CV_32FC4); }
+
+    copy_data(tmp, output, transpose);
+    return output;
+}
+
+static af::array array(const std::vector<cv::Mat>& input, bool transpose)
+{
+    af::array output;
+    try {
+        int h = input[0].rows;
+        int w = input[0].cols;
+        int c = input[0].channels();
+        if (transpose) {
+            output = af::array(h, w, c, input.size());
+        } else {
+            output = af::array(w, h, c, input.size());
+        }
+        for (unsigned i = 0; i < input.size(); i++) {
+            if(input[i].channels() != c)
+                throw af::exception("number of OpenCV matrix channels inconsistent");
+            af::array tmp = array(input[i], transpose);
+            output(af::span, af::span, af::span, i) = tmp;
+        }
+        return output;
+    } catch (const af::exception& ex) {
+        throw af::exception("std::vector<cv::Mat> to af::array error ");
+    }
+}
+
+
+/*
+ * arrayfire -> opencv
+ */
+
+// af::array conversion to cv::Mat
+static cv::Mat toMat(const af::array& input, int type, bool transpose)
+{
+    cv::Mat output;
+    const int channels = input.dims(2);
+    int ndims = input.numdims();
+    af::array input_;
+    if (transpose) {
+        if(channels == 1) {
+            input_ = input;
+        } else {
+            input_ = af::constant(0, channels, input.dims(1), input.dims(0));
+            for(int ii=0; ii<channels; ++ii) {
+                input_(ii, af::span, af::span) = af::moddims(input(af::span, af::span, ii), 1, input.dims(1), input.dims(0));
+            }
+        }
+        output = cv::Mat(input.dims(1), input.dims(0), CV_MAKETYPE(type, channels));
+    } else {
+        if(channels == 1) {
+            input_ = input.T();
+        } else {
+            input_ = af::constant(0, channels, input.dims(0), input.dims(1));
+            for(int ii=0; ii<channels; ++ii) {
+                input_(ii, af::span, af::span) = af::moddims(input(af::span, af::span, ii).T(), 1, input.dims(0), input.dims(1));
+            }
```

Comment:
This is just a reorder operation. Remove for loop. `array input_ = af::reorder(input, 2, 0, 1);`